src/main/com/algos/src/datastructures/caches/LRUCache.java

```diff
@@ -32,27 +21,27 @@ public LRUCache(int cap) {
 
     private void setCapacity(int newCapacity) {
         checkCapacity(newCapacity);
-        for (int i = data.size(); i > newCapacity; i--) {
-            Entry<K, V> evicted = evict();
-            data.remove(evicted.getKey());
-        }
         this.cap = newCapacity;
+        
     }
 
     private Entry<K, V> evict() {
         if (head == null) {
-            throw new RuntimeException("cache cannot be empty!");
+            throw new RuntimeException("Cache cannot be empty!");
         }
         Entry<K, V> evicted = head;
         head = evicted.getNextEntry();
-        head.setPreEntry(null);
+        if (head != null) {
+            head.setPreEntry(null);
+        }
         evicted.setNextEntry(null);
+        
         return evicted;
     }
 
     private void checkCapacity(int capacity) {
         if (capacity <= 0) {
-            throw new RuntimeException("capacity must greater than 0!");
+            throw new RuntimeException("Capacity must be greater than 0!");
```

Comment:
<div>

<strong>Bito Code Review Agent Run</strong> <a href=https://github.com/rishabhbohra3/Algo-tutorial-test/pull/6#issuecomment-2196728573>#cbf3aa</a><span> - 06/28/2024, 11:53 am</span>
<br>

<div><small>🔴</small> High importance</div>

<div id="suggestion">
<div id="issue">
<b>Issue:</b> The exception message change is trivial and does not address any critical issues. However, the method 'checkCapacity' throws a 'RuntimeException' which is a generic exception and may not be the best choice for indicating an invalid capacity. Using a more specific exception type would improve error handling and make the code more robust.
</div>
<div id="fix">
<b>Fix:</b> Replace the 'RuntimeException' with a more specific exception type, such as 'IllegalArgumentException', to better indicate that the issue is with the argument passed to the method.
</div>
<details>
<summary>
<b>Code suggestion</b>
</summary>
<div id="code">

```
 @@ -43,7 +43,7 @@ private void checkCapacity(int capacity) {
         if (capacity <= 0) {
 -            throw new RuntimeException("Capacity must be greater than 0!");
 +            throw new IllegalArgumentException("Capacity must be greater than 0!");
         }
     }
```
</div>
</details>
</div>
<br>


</div>